packages/airlock_digital/data_stream/execution_histories/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,383 @@
+      tag: json_event_original
+      target_field: json
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - append:
+      field: event.type
+      tag: append_info_into_event_type
+      value: info
+  - append:
+      field: event.category
+      tag: append_host_into_event_category
+      value: host
+  - set:
+      field: event.kind
+      tag: set_event_kind
+      value: event
+  - rename:
+      field: json.checkpoint
+      tag: rename_checkpoint
+      target_field: airlock_digital.execution_histories.checkpoint
+      ignore_missing: true
+  - set:
+      field: event.id
+      tag: set_event_id_from_execution_histories_checkpoint
+      copy_from: airlock_digital.execution_histories.checkpoint
+      ignore_empty_value: true
+  - rename:
+      field: json.commandline
+      tag: rename_commandline
+      target_field: airlock_digital.execution_histories.commandline
+      ignore_missing: true
+  - set:
+      field: process.command_line
+      tag: set_process_command_line_from_execution_histories_commandline
+      copy_from: airlock_digital.execution_histories.commandline
+      ignore_empty_value: true
+  - date:
+      field: json.datetime
+      tag: date_datetime
+      target_field: airlock_digital.execution_histories.datetime
+      formats:
+        - ISO8601
+      if: ctx.json?.datetime != null && ctx.json.datetime != ''
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - set:
+      field: '@timestamp'
+      tag: set_@timestamp_from_execution_histories_datetime
+      copy_from: airlock_digital.execution_histories.datetime
+      ignore_empty_value: true
+  - rename:
+      field: json.filename
+      tag: rename_filename
+      target_field: airlock_digital.execution_histories.filename
+      ignore_missing: true
+  - set:
+      field: file.path
+      tag: set_file_path_from_execution_histories_filename
+      copy_from: airlock_digital.execution_histories.filename
+      ignore_empty_value: true
+  - gsub:
+      field: file.path
+      pattern: "\\\\"
+      replacement: "/"
+      ignore_failure: true
+  - split:
+      field: file.path
+      separator: "\\/"
+      target_field: file.name
+      if: ctx?.file?.path != null
+      ignore_failure: true
+  - script:
+      tag: script_file_name
+      description: Set file.name to the last element of file.path.
+      lang: painless
+      source: |
+        ctx.file.name = ctx.file.name[ctx.file.name.length-1];
+      if: 'ctx?.file?.name !=null && ctx?.file?.name.length > 1'
```

```suggestion
      if: ctx.file?.name instanceof Array && ctx.file.name.length > 0
```
if this is only done when the array is longer than one, in the case of one, we end up with an array here. This normalises to the scalar if there is anything present.